apis/pipelines/v1alpha5/runconfiguration_conversion.go

```diff
@@ -1,3 +1,149 @@
 package v1alpha5
 
-func (*RunConfiguration) Hub() {}
+import (
+	"github.com/sky-uk/kfp-operator/apis/pipelines"
+	hub "github.com/sky-uk/kfp-operator/apis/pipelines/v1alpha6"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+)
+
+func (src *RunConfiguration) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*hub.RunConfiguration)
+
+	v1alpha6Remainder := hub.RunConfigurationConversionRemainder{}
+	if err := pipelines.GetAndUnsetConversionAnnotations(src, &v1alpha6Remainder); err != nil {
+		return err
+	}
+	dst.ObjectMeta = src.ObjectMeta
+	dst.Spec.Run.Pipeline = hub.PipelineIdentifier{
+		Name:    src.Spec.Run.Pipeline.Name,
+		Version: src.Spec.Run.Pipeline.Version,
+	}
+	dst.Spec.Run.ExperimentName = src.Spec.Run.ExperimentName
+	dst.Spec.Run.RuntimeParameters = convertRuntimeParametersTo(
+		src.Spec.Run.RuntimeParameters,
+	)
+	dst.Spec.Run.Artifacts = convertArtifactsTo(src.Spec.Run.Artifacts)
+	dst.Spec.Triggers = convertTriggersTo(src.Spec.Triggers, v1alpha6Remainder)
+	dst.Status = hub.RunConfigurationStatus{
+		ObservedPipelineVersion: src.Status.ObservedPipelineVersion,
+		SynchronizationState:    src.Status.SynchronizationState,
+		Provider:                src.Status.Provider,
+		ObservedGeneration:      src.Status.ObservedGeneration,
+	}
+	return nil
+}
+
+func (dst *RunConfiguration) ConvertFrom(srcRaw conversion.Hub) error {
+	src := srcRaw.(*hub.RunConfiguration)
+	v1alpha6Remainder := hub.RunConfigurationConversionRemainder{}
+	dst.ObjectMeta = src.ObjectMeta
+	dst.Spec.Run.Pipeline = PipelineIdentifier{
+		Name:    src.Spec.Run.Pipeline.Name,
+		Version: src.Spec.Run.Pipeline.Version,
+	}
+	dst.Spec.Run.ExperimentName = src.Spec.Run.ExperimentName
+	dst.Spec.Run.RuntimeParameters = convertRuntimeParametersFrom(src.Spec.Run.RuntimeParameters)
+	dst.Spec.Run.Artifacts = convertArtifactsFrom(src.Spec.Run.Artifacts)
+	dst.Spec.Triggers = convertTriggersFrom(src.Spec.Triggers, &v1alpha6Remainder)
+	dst.Status = RunConfigurationStatus{
+		ObservedPipelineVersion: src.Status.ObservedPipelineVersion,
+		SynchronizationState:    src.Status.SynchronizationState,
+		Provider:                src.Status.Provider,
+		ObservedGeneration:      src.Status.ObservedGeneration,
+	}
+	return pipelines.SetConversionAnnotations(dst, &v1alpha6Remainder)
+}
+
+// Converts spoke Triggers into hub Triggers whilst taking into account of
+// annotations that convey hub Triggers (remainder).
+//
+// If the spoke Schedule matches a CronExpression in the remainder then the
+// conversion will use the StartTime and EndTime ptrs from the remainder.
```

Comment:
```suggestion
// conversion will use the StartTime and EndTime pointers from the remainder.
```

Reply:
done in 1060cf492503107798674c9aa4752556a45ed919